### ISSUE TITLE
The REST API is not working with the current code at master

### ISSUE THREAD

**Issue by judavi** [grafeas/grafeas#389]
## Expected Behavior

The REST API should list the projects available using the REST endpoint /v1beta/projects

## Actual Behavior

The REST API is replying with an error:
```
{"error":"all SubConns are in TransientFailure, latest connection error: timed out waiting for server handshake","message":"all SubConns are in TransientFailure, latest connection error: timed out waiting for server handshake","code":14,"details":[]}%
```

## Steps to Reproduce the Problem

1.  Download the master branch
2.  Execute 
```
make build 
```
3.  Go to v1beta1/main
```
cd $GOPATH/src/github.com/grafeas/grafeas/go/v1beta1/main
```
4.  Create a config file for connecting with a local Postgres database
```
cat config.yaml
grafeas:
  api:
    address: "localhost:8080"
    cafile:
    keyfile:
    certfile:
    cors_allowed_origins:
  storage_type: "postgres"
  postgres:
    host: "0.0.0.0:5432"
    dbname: "postgres"
    user: "postgres"
    password: "password"
    sslmode: "disable"
    paginationkey:

```
5. Run main.go using the config file using GRPC variables to get more information 
```
GRPC_GO_LOG_VERBOSITY_LEVEL=99 GRPC_GO_LOG_SEVERITY_LEVEL=info GRPC_VERBOSITY=DEBUG go run main.go --config config.yaml
```
6.  Execute a GET call to the projects API
```
curl http://localhost:8080/v1beta1/projects
```

## Excepted output
```
{"projects":[],"nextPageToken":""}
```

## Current output client 
```
{"error":"all SubConns are in TransientFailure, latest connection error: timed out waiting for server handshake","message":"all SubConns are in TransientFailure, latest connection error: timed out waiting for server handshake","code":14,"details":[]}%
```

## Current output server
```
INFO: 2019/09/16 20:43:56 parsed scheme: ""
INFO: 2019/09/16 20:43:56 scheme "" not registered, fallback to default scheme
INFO: 2019/09/16 20:43:56 ccResolverWrapper: sending update to cc: {[{[::]:8080 0  <nil>}] <nil>}
INFO: 2019/09/16 20:43:56 ClientConn switching balancer to "pick_first"
2019/09/16 20:43:56 grpc server is configured without client certificate authentication
INFO: 2019/09/16 20:43:56 pickfirstBalancer: HandleSubConnStateChange: 0xc000032030, CONNECTING
WARNING: 2019/09/16 20:44:16 grpc: addrConn.createTransport failed to connect to {[::]:8080 0  <nil>}: didn't receive server preface in time. Reconnecting...
INFO: 2019/09/16 20:44:16 pickfirstBalancer: HandleSubConnStateChange: 0xc000032030, TRANSIENT_FAILURE
INFO: 2019/09/16 20:44:16 transport: loopyWriter.run returning. connection error: desc = "transport is closing"
```

## Extra information

This situation is not happening with old versions of the code. If you run the same scenario with the #361 the REST API starts working again.
Some extra info could be also found in https://github.com/grafeas/grafeas/issues/381

**Comment by john-tipper** [comment]
I can confirm that this is an issue with `memstore` too:

```
docker run -p 8080:8080  us.gcr.io/grafeas/grafeas-server:v0.1.2
```

In a separate terminal:

```
curl http://localhost:8080/v1beta1/projects
```

```
{"error":"all SubConns are in TransientFailure, latest connection error: timed out waiting for server handshake","message":"all SubConns are in TransientFailure, latest connection error: timed out waiting for server handshake","code":14,"details":[]}
```

The bug exists for version `v0.1.2` and `v0.1.1`, but not for `v0.1.0`.
There was a bug created

**Comment by judavi** [comment]
@john-tipper thanks for your help confirming the issue exists! I was getting frustrated that I was the only one having the issue.

**Comment by judavi** [comment]
I think something is broken on the use of grpc-gateway but my Go knowledge is still no good enough to identify the main reason. Any suggestion @aysylu @vtsao ?

**Comment by aysylu** [comment]
@judavi sorry you're still having issues with this and thanks for your patience with this while we figure it out!

@john-tipper thank you for helping narrow down this issue! I'm looking over the commits between `v0.1.0` and `v0.1.1` and nothing jumps out as the cause yet, but I'm continuing to dig into this.

@judavi could you please run `curl -v` and paste the output here? I suspect the server is somehow requiring TLS. Do you see `grpc server is configured without client certificate authentication` anywhere in the logs?

**Comment by aysylu** [comment]
@judavi I'm prioritizing to look into this early next week. Stay tuned and keep me posted on your debugging, if you work on it in the meantime. Thanks again for reporting this and your patience!

**Comment by judavi** [comment]
thanks @aysylu ! I'll post my updates if I found something else.

**Comment by aysylu** [comment]
@judavi unfortunately, I haven't yet been able to track down the issue, partially due to my split attention with oncall duties this week. I'm getting someone else on my team to take a look as well, in case my workload prevents me from looking into this in the nearest future. Please stay tuned and thanks for your patience!

**Comment by grafeas-monitor** [comment]
+noahklein, who will pick up this work.

**Comment by noahklein** [comment]
Closing this issue, as it appears to have been fixed by #395 (thanks, @judavi!) Please reopen if you're still encountering this problem.
